api/src/middleware/error-handler.ts

```diff
@@ -1,105 +1,124 @@
-import { ErrorCode, MethodNotAllowedError, isDirectusError } from '@directus/errors';
-import { isObject, toArray } from '@directus/utils';
+import { ErrorCode, isDirectusError } from '@directus/errors';
+import type { DeepPartial } from '@directus/types';
+import { isObject } from '@directus/utils';
 import { getNodeEnv } from '@directus/utils/node';
-import type { ErrorRequestHandler } from 'express';
+import type { ErrorRequestHandler, NextFunction, Request, Response } from 'express';
 import getDatabase from '../database/index.js';
 import emitter from '../emitter.js';
 import { useLogger } from '../logger.js';
 
-// Note: keep all 4 parameters here. That's how Express recognizes it's the error handler, even if
-// we don't use next
-const errorHandler: ErrorRequestHandler = (err, req, res, _next) => {
-	const logger = useLogger();
-
-	let payload: any = {
-		errors: [],
+type ApiError = {
+	message: string;
+	extensions: {
+		code: string;
+		[key: string]: any;
 	};
+};
 
-	const errors = toArray<unknown>(err);
+const FALLBACK_ERROR = {
+	status: 500,
+	message: 'An unexpected error occurred.',
+	extensions: {
+		code: 'INTERNAL_SERVER_ERROR',
+	},
+};
```

Comment:
Could we move this error definition to the `@directus/errors` package? 😄

Reply:
I didn't do so because we don't need a fully fledged `DirectusError` here and `DirectusError` would expect `code` under its root, so we would have to some re-transformation.
But I agree that it would make sense to make it available to clients via `@directus/errors`.
I'll have a look 👍

Reply:
Yeah exactly for extension use was my initial thinking there 😄

Reply:
Done ✅